diff --git a/tests/test_bes_pipenv.sh b/tests/test_bes_pipenv.sh
index eea4302..3fa12e4 100755
--- a/tests/test_bes_pipenv.sh
+++ b/tests/test_bes_pipenv.sh
@@ -53,10 +53,14 @@ function test_bes_pipenv_ensure()
   fi
   local _tmp=/tmp/test_bes_pipenv_ensure_$$
 
+  bes_assert "[[ $(bes_testing_call_function bes_pipenv_has_pipenv ${_builtin_python} ${_tmp} ) == 1 ]]"
+  
   BES_PIP_EXTRA_ARGS="--no-cache-dir" bes_pipenv_ensure "${_builtin_python}" "${_tmp}" 20.2.2 2020.8.13
   local _ensure_rv=$?
   bes_assert "[[ ${_ensure_rv} == 0 ]]"
 
+  bes_assert "[[ $(bes_testing_call_function bes_pipenv_has_pipenv ${_builtin_python} ${_tmp} ) == 0 ]]"
+  
 #####  
 ######  bes_assert "[[ $(bes_testing_call_function bes_pip_has_pip ${_builtin_python} ) == 1 ]]"
 #####
@@ -100,7 +104,8 @@ function test_bes_pipenv_call()
   local _ensure_rv=$?
   bes_assert "[[ ${_ensure_rv} == 0 ]]"
 
-  bes_pipenv_call "${_builtin_python}" "${_tmp}" --version
+  local _pipenv_version=$(bes_pipenv_call "${_builtin_python}" "${_tmp}" --version)
+  bes_assert "[[ ${_pipenv_version} == 2020.8.13 ]]"
   
   rm -rf ${_tmp}
 }
